lib/Resque/Job.php

```diff
@@ -230,7 +230,12 @@ public function fail($exception)
 			$this->worker,
 			$this->queue
 		);
-		Resque_Stat::incr('failed');
+
+        if (Resque_Failure::getBackend() == 'Resque_Failure_RedisMultipleQueues') {
+            Resque_Stat::incr($this->queue . '_failed');
+        } else {
+            Resque_Stat::incr('failed');
+        }
```

Comment:
We could probably roll with putting it into the failure backend for now and if someone upstream complains (with a good reason!) we change it back then.

However, there is minimal likelihood of this being seen, let alone accepted by someone with commit rights to upstream php-resque.